platform/pulp/download/batch.py

```diff
@@ -0,0 +1,535 @@
+from concurrent.futures import ThreadPoolExecutor
+from gettext import gettext as _
+from logging import getLogger, DEBUG
+from queue import Queue, Empty
+from threading import Thread, current_thread
+
+from .single import Context
+
+
+# The default queuing backlog.
+BACKLOG = 1024
+# The default concurrency.
+CONCURRENT = 10
```

Comment:
I think our current default of 5 is a lot and plenty to maximize available resources. Katello even reduces it to 4 via an importer setting. Beyond about 3, the returns diminish rapidly, and resource contention ramps up.